build/cmake/GenerateUninstaller.cmake

```diff
@@ -258,13 +210,37 @@ append_line("${init}")
 # Generate a "remove a file" command
 function(add_rmfile filename)
     file(TO_NATIVE_PATH "${filename}" native)
-    append_line("${__rmfile} '${native}'")
+    if(WIN32)
+      set(file "%__prefix%\\${native}")
+      set(rmfile_lines
+        "  <nul set /p \"=Remove file: ${file} \""
+        "  if EXIST \"${file}\" ("
+        "    del /Q /F \"${file}\" && echo - ok"
+        "  ) ELSE echo - skipped: not present"
+        ") && (")
+      string(REPLACE ";" "\n" rmfile "${rmfile_lines}")
+      append_line("${rmfile}")
+    else()
+      append_line("__rmfile '${native}'")
+    endif()
 endfunction()
 
 # Generate a "remove a directory" command
 function(add_rmdir dirname)
     file(TO_NATIVE_PATH "${dirname}" native)
-    append_line("${__rmdir} '${native}'")
+    if(WIN32)
+      set(dir "%__prefix%\\${native}")
+      set(rmdir_lines
+        "  <nul set /p \"=Remove directory: ${dir} \""
```

Comment:
> such as the inability to use `set`

No change requested. But I expect using `set` in the caller context appears OK. I can use `set` in a command window without running a batch script. I expect the other `echo` commands could be revised to use this pattern, but see no reason to.

Reply:
I may have mixed up "`call` has no effect" with "`set` doesn't work" during prior testing when attempting to use labels. That being said, I think we should refrain from setting any more variables than is strictly necessary given this is being evaluated in the caller's context.